services/src/main/java/org/keycloak/services/Urls.java

```diff
@@ -273,4 +277,33 @@ private static UriBuilder themeBase(URI baseUri) {
     public static URI samlRequestEndpoint(final URI baseUri, final String realmName) {
         return realmBase(baseUri).path(RealmsResource.class, "getProtocol").build(realmName, SamlProtocol.LOGIN_PROTOCOL);
     }
+
+    public static void checkUrl(KeycloakSession session, String url, String name) throws IllegalArgumentException{
+        if (url == null) {
+            return;
+        }
+
+        URL parsed;
+
+        try {
+            parsed = new URL(url);
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException("The url [" + name + "] is malformed", e);
+        }
+
+        String protocol = parsed.getProtocol().toLowerCase();
+
+        if (!protocol.startsWith("http")) {
```

Comment:
Should this not be an exact match on http or https? Otherwise the accepted protocol could be "httpasdfsadfsafdsadfasdfsafdsadfsafdsadf".

Reply:
How you would even accept requests for a protocol that is unknown by the server? It would be either `http` or `https`, right?

Reply:
Forget about it. This is about validating input and not incoming connections.